### PR TITLE
MODLD-238 fix marc deduplication data

### PR DESCRIPTION
**Pull request by aleksei-pronichev**
- add possibility to have same data in different fields

### CODE REVIEW

File: src/test/resources/expected_marc.jsonl [hunk 1]
```diff
@@ -498,6 +510,20 @@
     "506" : {
       "subfields" : [ {
         "a" : "terms, jurisdiction, provisions, users, authorization, terminology, date, agency"
+      }, {
+        "b" : "terms, jurisdiction, provisions, users, authorization, terminology, date, agency"
```
[the remark this answers is not in the document]
Comment (pkjacob) [reply src/test/resources/expected_marc.jsonl:514]: That is correct. In this case, we need only `$a` in the marc record.

[the remark this answers is not in the document]
Comment (aleksei-pronichev) [reply src/test/resources/expected_marc.jsonl]: outdated, new version is ready

**Review by PBobylev — APPROVED** [review]

File: src/main/java/org/folio/marc4ld/configuration/Bibframe4MarcRulesConfig.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+import org.folio.marc4ld.configuration.property.Marc4BibframeRules;
+import org.folio.marc4ld.service.condition.ConditionChecker;
+import org.folio.marc4ld.service.dictionary.DictionaryProcessor;
+import org.folio.marc4ld.service.ld2marc.field.Bibframe2MarcFieldRule;
+import org.folio.marc4ld.service.ld2marc.field.impl.Bibframe2MarcFieldRuleImpl;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+@Configuration
+public class Bibframe4MarcRulesConfig {
```
Comment (askhat-abishev) [inline src/main/java/org/folio/marc4ld/configuration/Bibframe4MarcRulesConfig.java:20]: just not to forget rename to Bibframe2MarcRulesConfig

**Review by askhat-abishev — APPROVED** [review]

File: src/main/java/org/folio/marc4ld/service/ld2marc/field/Bibframe2MarcFieldRule.java [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+import org.folio.marc4ld.service.ld2marc.field.param.ControlFieldParameter;
+import org.folio.marc4ld.service.ld2marc.field.param.SubFieldParameter;
+
+public interface Bibframe2MarcFieldRule {
+
+  String getTag();
+
+  boolean isSuitable(Resource resource, PredicateDictionary predicate);
+
+  boolean isSatisfied(Resource resource);
```
Comment (pkjacob) [inline src/main/java/org/folio/marc4ld/service/ld2marc/field/Bibframe2MarcFieldRule.java:15]: This method returns a boolean value to indicate if DataField can be created out of Resouce. So, I think `canCreateDataField` or `isDataFieldCreatable` may be a better name.

File: src/main/java/org/folio/marc4ld/service/ld2marc/field/impl/Bibframe2MarcFieldRuleImpl.java [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+    );
+    this.indProperty2 = mapToIndicator(
+      Marc4BibframeRules.FieldRule::getInd2,
+      Marc4BibframeRules.Marc2ldCondition::getInd2
+    );
+  }
+
+  @Override
+  public boolean isSuitable(Resource resource, PredicateDictionary predicate) {
+    return Objects.equals(fieldRule.getTypes(), resource.getTypeNames())
```
Comment (pkjacob) [inline src/main/java/org/folio/marc4ld/service/ld2marc/field/impl/Bibframe2MarcFieldRuleImpl.java:61]: Future enhancement that we need to implement -

Once we add "Lookup" feature in Marva UI, a work's "subject" could point to either one of the following 2 resources
  >> A resource with types `Concept`, `Person`
  >> A resource with type just `Person`

In both cases, we have to transform the resource back to marc field `600`. 
Due to the check `Objects.equals(fieldRule.getTypes(), resource.getTypeNames())`, marc 600 will not be created for subjects with type just `Person`. Something to fix in future.

@PBobylev  @aleksei-pronichev  @askhat-abishev

**Review by pkjacob — APPROVED** [review]

File: src/main/java/org/folio/marc4ld/service/ld2marc/field/impl/Bibframe2MarcFieldRuleImpl.java (hunk 4 above)
Comment (aleksei-pronichev) [reply src/main/java/org/folio/marc4ld/service/ld2marc/field/impl/Bibframe2MarcFieldRuleImpl.java:61]: Great note, move it to the chat

**Pull request merged**